src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Windows.cs

```diff
@@ -410,55 +410,61 @@ private static ModuleInfo[] GetModuleInfos(int processId, bool firstModuleOnly)
                     if (moduleCount <= moduleHandles.Length) break;
                     moduleHandles = new IntPtr[moduleHandles.Length * 2];
                 }
-                List<ModuleInfo> moduleInfos = new List<ModuleInfo>();
 
-                int ret;
+                List<ModuleInfo> moduleInfos = new List<ModuleInfo>(moduleCount);
+                StringBuilder baseName = new StringBuilder(1024);
+                StringBuilder fileName = new StringBuilder(1024);
+
                 for (int i = 0; i < moduleCount; i++)
                 {
-                    try
+                    while (true)
```

Comment:
>I don't understand the purpose of this while loop. Is it just to enable those break;s in the body?

exactly

>If so, wouldn't it be better to use continue; instead of break; and get rid of the while (true)?

That will require move the rest of the loop to beginning and add one more check (`i == 1`)

	baseName.Clear();
	fileName.Clear();

	// If the user is only interested in the main module, break now.
	// This avoid some waste of time. In addition, if the application unloads a DLL                     
	// we will not get an exception. 
	if (firstModuleOnly && i == 1) { break; }

and Clear() foк SB will be invoked at first iteration.
It's okay?